iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/GcpIpRestrictedAdminAccessCheckPart.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.terraform.checks.gcp;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.checks.TextUtils;
+import org.sonar.iac.terraform.api.tree.ExpressionTree;
+import org.sonar.iac.terraform.checks.IpRestrictedAdminAccessCheck;
+import org.sonar.iac.terraform.checks.ResourceVisitor;
+
+import static org.sonar.iac.terraform.checks.IpRestrictedAdminAccessCheck.ALL_IPV4;
+import static org.sonar.iac.terraform.checks.IpRestrictedAdminAccessCheck.ALL_IPV6;
+import static org.sonar.iac.terraform.checks.IpRestrictedAdminAccessCheck.MESSAGE;
+import static org.sonar.iac.terraform.checks.IpRestrictedAdminAccessCheck.SECONDARY_MSG;
+
+public class GcpIpRestrictedAdminAccessCheckPart extends ResourceVisitor {
+
+  private static final Set<String> SENSITIVE_PREFIXES = Set.of(ALL_IPV4, ALL_IPV6, "0::0/0", "::0/0");
+
+  private static final Predicate<ExpressionTree> RANGE_CONTAINS_SENSITIVE_PORTS =
+    range -> TextUtils.matchesValue(range, IpRestrictedAdminAccessCheck::rangeContainsSshOrRdpPort).isTrue();
+
+  private static final Predicate<ExpressionTree> SENSITIVE_IP_RANGE =
+    range -> TextUtils.matchesValue(range, SENSITIVE_PREFIXES::contains).isTrue();
```

Comment:
(a hint you probably don't need): this is reducible to
`treePredicate(SENSITIVE_PREFIXES::contains)`